### PR TITLE
MAISTRA-1192 & MAISTRA-1193 Add mutating webhooks for SMMR and SMCP

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

File: pkg/controller/servicemesh/webhooks/server.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	operatorNamespace := common.GetOperatorNamespace()
+	hookServer, err := webhook.NewServer(
+		componentName,
+		mgr,
+		webhook.ServerOptions{
+			Port:    11999,
+			CertDir: "/tmp/cert",
+			BootstrapOptions: &webhook.BootstrapOptions{
+				ValidatingWebhookConfigName: fmt.Sprintf("%s.servicemesh-resources.maistra.io", operatorNamespace),
+				MutatingWebhookConfigName:   fmt.Sprintf("%s.servicemesh-resources.maistra.io", operatorNamespace),
```
Comment (dgn) [inline pkg/controller/servicemesh/webhooks/server.go:39]: I guess this was here before but I'm not sure I understand- what is this ConfigName about?

**Review by dgn — APPROVED** [review]
Looks good.

I'm assuming removing the code that e.g. removes the control plane namespaces from memberrolls in the controller will come with a later PR?

**Review by rcernich — APPROVED** [review]

File: pkg/controller/servicemesh/webhooks/server.go (hunk 1 above)
Comment (luksa) [reply pkg/controller/servicemesh/webhooks/server.go:39]: That's the name of the ValidatingWebhookConfiguration object that the operator creates.

**Comment by luksa** [comment]
> Looks good.
> 
> I'm assuming removing the code that e.g. removes the control plane namespaces from memberrolls in the controller will come with a later PR?

Yes. We can do that later when we also remove the if statements that skip the control plane namespace.

**Pull request merged**